x-pack/plugins/enterprise_search/common/types/connectors.ts

```diff
@@ -38,11 +38,68 @@ export interface IngestPipelineParams {
   run_ml_inference: boolean;
 }
 
+export enum FilteringPolicy {
+  EXCLUDE = 'exclude',
+  INCLUDE = 'include',
+}
+
+export enum FilteringRuleRule {
+  CONTAINS = 'contains',
+  ENDS_WITH = 'ends_with',
+  EQUALS = 'equals',
+  GT = '>',
+  LT = '<',
+  REGEX = 'regex',
+  STARTS_WITH = 'starts_with',
+}
+
+export interface FilteringRule {
+  created_at: string;
+  field: string;
+  id: string;
+  order: number;
+  policy: FilteringPolicy;
+  rule: FilteringRuleRule;
+  updated_at: string;
+  value: string;
+}
+
+export interface FilteringValidation {
+  ids: string[];
+  messages: string[];
+}
+
+export enum FilteringValidationState {
+  EDITED = 'edited',
+  INVALID = 'invalid',
+  VALID = 'valid',
+}
+
+export interface FilteringRules {
+  advanced_snippet: {
+    created_at: string;
+    updated_at: string;
+    value: Record<string, unknown>;
+  };
+  rules: FilteringRule[];
+  validation: {
+    errors: FilteringValidation[];
+    state: FilteringValidationState;
+  };
+}
+
+export interface FilteringConfig {
+  active: FilteringRules;
+  domain: string;
+  draft: FilteringRules;
+}
+
 export interface Connector {
   api_key_id: string | null;
   configuration: ConnectorConfiguration;
   description: string | null;
   error: string | null;
+  filtering: FilteringConfig[];
```

Comment:
`| null`?

Reply:
Ha! That was my first approach too. Sander didn't like it. :) https://github.com/elastic/kibana/pull/144191#discussion_r1009194820